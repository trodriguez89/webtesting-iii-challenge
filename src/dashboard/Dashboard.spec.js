import React from 'react';
import {render, fireEvent} from "@testing-library/react";

import Dashboard from "./Dashboard";

// Test away

test("Dashboard matches snapshot", () => {
    const tree = render(<Dashboard />)
    expect(tree.baseElement).toMatchSnapshot();

})

test("Display after lock", () => {
    const { getByText } = render(<Dashboard />)

    fireEvent.click(getByText(/close gate/i))
    fireEvent.click(getByText(/lock gate/i))
    getByText(/closed/i)
    getByText(/locked/i)
    getByText(/open gate/i)
    getByText(/unlock gate/i)
});

test("Display after clicking lock gate", () => {
    const { getByText } = render(<Dashboard />)

    fireEvent.click(getByText(/close gate/i))
    fireEvent.click(getByText(/lock gate/i))
    getByText(/closed/i)
    getByText(/unlock gate/i)
    getByText(/open gate/i)
    getByText(/lock gate/i)
});

test("Displays current display after clicking lock gate", () => {
    const { getByText } = render(<Dashboard />)

    fireEvent.click(getByText(/close gate/i))
    fireEvent.click(getByText(/lock gate/i))
    fireEvent.click(getByText(/unlock gate/i))
    fireEvent.click(getByText(/open gate/i))
    getByText(/open/i)
    getByText(/unlocked/i)
    getByText(/close gate/i)
    getByText(/lock gate/i)
});
