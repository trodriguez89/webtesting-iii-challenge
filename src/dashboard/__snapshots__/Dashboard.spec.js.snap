// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard matches snapshot 1`] = `
<body>
  <div>
    <div
      class="display panel"
    >
      <div
        class="led green-led"
      >
        Unlocked
      </div>
      <div
        class="led green-led"
      >
        Open
      </div>
    </div>
    <div
      class="controls panel"
    >
      <button
        class="toggle-btn"
        data-testid="locked"
        disabled=""
      >
        Lock Gate
      </button>
      <button
        class="toggle-btn"
        data-testid="closed"
      >
        Close Gate
      </button>
    </div>
  </div>
</body>
`;
